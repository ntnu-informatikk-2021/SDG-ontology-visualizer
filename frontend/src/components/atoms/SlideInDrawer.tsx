import React from 'react';
import { Box } from '@chakra-ui/react';
import { AnimatePresence } from 'framer-motion';
import { MotionBox } from '../../types/react/componentTypes';

interface SlideInDrawerProps {
  expanded: Boolean;
  width: string;
  children: JSX.Element;
}

const SlideInDrawer: React.FC<SlideInDrawerProps> = ({
  expanded,
  width,
  children,
}: SlideInDrawerProps) => (
  <AnimatePresence initial={false}>
    {expanded && (
      <MotionBox
        key="content"
        initial="collapsed"
        animate="open"
        exit="collapsed"
        variants={{
          open: { width: 'auto' },
          collapsed: { width: 0 },
        }}
        transition={{ duration: 0.3, ease: 'easeInOut' }}
      >
        <Box overflow="hidden">
          <Box w={width}>{children}</Box>
        </Box>
      </MotionBox>
    )}
  </AnimatePresence>
);

export default SlideInDrawer;
