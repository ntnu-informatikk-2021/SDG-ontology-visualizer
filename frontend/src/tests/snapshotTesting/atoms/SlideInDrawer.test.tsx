import React from 'react';
import renderer from 'react-test-renderer';

import SlideInDrawer from '../../../components/atoms/SlideInDrawer';

it('renders when there are no items', () => {
  const tree = renderer
    .create(
      <SlideInDrawer expanded width="10px">
        <></>
      </SlideInDrawer>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
