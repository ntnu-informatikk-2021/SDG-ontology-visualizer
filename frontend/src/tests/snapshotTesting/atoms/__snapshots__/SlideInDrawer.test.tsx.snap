// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there are no items 1`] = `
<div
  className="css-0"
  style={
    Object {
      "width": "auto",
    }
  }
>
  <div
    className="css-i6bazn"
  >
    <div
      className="css-bosr71"
    />
  </div>
</div>
`;
